### PR TITLE
Fix `clippy::only-used-in-recursion`

### PR DESCRIPTION
**Pull request by cmichi**
`master` CI with latest nightly fails at https://gitlab.parity.io/parity/ink/-/jobs/1440908 with https://rust-lang.github.io/rust-clippy/master/index.html#only_used_in_recursion.

### CODE REVIEW

**Comment by paritytech-cicd-pr** [comment]
## 🦑 📈 ink! Example Contracts ‒ Changes Report 📉 🦑
 These are the results when building the `examples/*` contracts from this branch with `cargo-contract 0.18.0-9362c04` and comparing them to ink! `master`: 

|                          | Δ Optimized Size | Δ Used Gas | Total Optimized Size | Total Used Gas |
|:--------------------------|------------------:|------------:|----------------------:|----------------:|
| `accumulator`            |                  |            | 1.03 K               |                |
| `adder`                  |                  |            | 2.17 K               |                |
| `contract-introspection` |                  |            | 2.37 K               |                |
| `contract-terminate`     |                  |            | 0.94 K               | 214_418        |
| `contract-transfer`      |                  |            | 8.15 K               | 14_418         |
| `delegator`              |                  |            | 6.37 K               | 46_295         |
| `dns`                    |                  |            | 8.84 K               | 43_254         |
| `erc1155`                |                  |            | 17.23 K              | 86_508         |
| `erc20`                  |                  |            | 8.49 K               | 43_254         |
| `erc721`                 |                  |            | 11.81 K              | 115_344        |
| `flipper`                |                  |            | 1.31 K               | 14_418         |
| `incrementer`            |                  |            | 1.21 K               | 14_418         |
| `multisig`               |                  |            | 25.20 K              | 93_527         |
| `proxy`                  |                  |            | 2.90 K               | 29_536         |
| `rand-extension`         |                  |            | 3.92 K               | 14_418         |
| `subber`                 |                  |            | 2.19 K               |                |
| `trait-erc20`            |                  |            | 8.76 K               | 43_254         |
| `trait-flipper`          |                  |            | 1.00 K               | 14_418         |
| `trait-incrementer`      |                  |            | 1.19 K               | 28_836         |
| `upgradeable-contract`   |                  |            | 2.99 K               | 14_950         |
| `upgradeable-flipper`    |                  |            | 1.55 K               |                |



[Link to the run](https://gitlab.parity.io/parity/ink-waterfall/-/pipelines/183835) | Last update: Wed Mar 16 08:40:24 CET 2022

File: crates/env/src/types.rs [hunk 1]
```diff
@@ -57,44 +57,44 @@ impl FromLittleEndian for u8 {
     type Bytes = [u8; 1];
 
     #[inline]
-    fn from_le_bytes(bytes: Self::Bytes) -> Self {
-        Self::from_le_bytes(bytes)
+    fn from_le_bytes(_bytes: Self::Bytes) -> Self {
+        Self::from_le_bytes(_bytes)
```
Comment (ascjones) [inline crates/env/src/types.rs:61]: Only this `u8` impl triggers the warning.

**Review by ascjones — COMMENTED** [review]
Looks like a clippy bug, because that is calling the inherent `from_le_bytes` method on the uint. Curious that it just happens for the `u8`. Perhaps could replace that with `u8::from_le_bytes` to be more explicit? The underscore could be confusing.

**Comment by cmichi** [comment]
>Perhaps could replace that with u8::from_le_bytes to be more explicit?

Did it everywhere, fixes it. I know that it's only for u8, but for consistency I changed it everywhere.

>The underscore could be confusing.

I agree.

**Review by ascjones — APPROVED** [review]

**Pull request merged**